string name
int64 timestamp
int32 encoder

